internal/resources/backupschedule/resource_backup_schedule.go

```diff
@@ -507,6 +507,11 @@ func readCGResourceWait(ctx context.Context, config *authctx.TanzuContext,
 			resourceFullName)
 
 		if err != nil || resp == nil || resp.Schedule == nil {
+			if clienterrors.IsUnauthorizedError(err) {
+				authctx.RefreshUserAuthContext(config, clienterrors.IsUnauthorizedError, err)
+				continue
+			}
+
 			return nil, err
```

Comment:
@rnarenpujari, can we add a unit test for this?

Reply:
Currently we only have acceptance tests and simulating this condition seems difficult. Looks like other resources also have mock tests which would be ideal here - but it doesn't seem the contractors added those when writing the data protection resources. Since it might be a sizeable effort, would prefer to track this separately. But let me know if you feel strongly about adding them with this PR itself.

Reply:
Will track the addition of mock tests for this resource separately.